src/math/matrix.c

```diff
@@ -3,93 +3,112 @@
 // Copyright(c) 2022 Intel Corporation. All rights reserved.
```

Comment:
regarding your commit message, please use atomic commits, this should be a chain of four different patches, each one implementing the change your reference, makes review a lot easier.

Reply:
I verified the content of commit `19db9c9c949de96b9b3dd93e08380dd825edb56c` to ensure tight coupling and functional improvement. I attempted a rebase to investigate logical segmentation, but discovered that it resulted in a single, atomic change to the code. I'm open to further discussion and willing to help clarify aspects of the commit to make the review process go more smoothly.

Reply:
Yes, you can and should split this into separate commits. Some changes add missing functions documents, some changes add further errors checks, and some changes optimize the code of two separate functions. Those are independent from each other can be in separate commits.

Reply:
Done. Please take a look.